packages/next/build/jest/jest.ts

```diff
@@ -88,6 +95,19 @@ export default function nextJest(options: { dir?: string } = {}) {
           ? await customJestConfig()
           : customJestConfig) ?? {}
 
+      // Setup transforms:
+      if (options.experimentalEsm === true && !isEsmProject) {
```

Comment:
This warning may not be necessary, but could help if the developer hasn't actually configured the `package.json` to have `type: "module"` present. It's not strictly necessary.